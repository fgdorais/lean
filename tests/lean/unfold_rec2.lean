open nat

definition f (a n : ℕ) : ℕ := nat.rec_on n (λa', a') (λn' f' a', f' (a' * 2)) a

example (a n : nat) : a = 2 → f a (succ n) = f 4 n :=
begin
  unfold f at {1},
  state,
  intros, subst a
end
