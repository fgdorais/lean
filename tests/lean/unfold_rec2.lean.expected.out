unfold_rec2.lean:8:2: proof state
a n : ℕ
⊢ a = 2 → f (a * 2) n = f 4 n
